Content.Server/Body/Systems/RespiratorSystem.cs

```diff
@@ -41,155 +56,95 @@ public override void Update(float frameTime)
 
                 respirator.AccumulatedFrametime += frameTime;
 
-                if (respirator.AccumulatedFrametime < 1)
-                {
+                if (respirator.AccumulatedFrametime < respirator.CycleDelay)
                     continue;
-                }
-
-                ProcessGases(uid, respirator, blood, body);
+                respirator.AccumulatedFrametime -= respirator.CycleDelay;
+                UpdateSaturation(respirator.Owner, -respirator.CycleDelay, respirator);
 
-                respirator.AccumulatedFrametime -= 1;
+                switch (respirator.Status)
+                {
+                    case RespiratorStatus.Inhaling:
+                        Inhale(uid, body);
+                        respirator.Status = RespiratorStatus.Exhaling;
+                        break;
+                    case RespiratorStatus.Exhaling:
+                        Exhale(uid, body);
+                        respirator.Status = RespiratorStatus.Inhaling;
+                        break;
+                }
 
-                if (SuffocatingPercentage(respirator) > 0)
+                if (respirator.Saturation < respirator.SuffocationThreshold)
                 {
+                    if (_gameTiming.CurTime >= respirator.LastGaspPopupTime + respirator.GaspPopupCooldown)
+                    {
+                        respirator.LastGaspPopupTime = _gameTiming.CurTime;
+                        _popupSystem.PopupEntity(Loc.GetString("lung-behavior-gasp"), uid, Filter.Pvs(uid));
+                    }
+
+                    respirator.SuffocationCycles += 1;
                     TakeSuffocationDamage(uid, respirator);
                     continue;
                 }
 
+                respirator.SuffocationCycles = 0;
                 StopSuffocation(uid, respirator);
```

Comment:
The way that this is currently done means that when a person is partially suffocating (e.g., 16% oxygen mix in a tank), so that during the breathing cycles the saturation does something like 3 -> 1 -> -1 -> ~3, then they alternate between suffocating and not suffocating which spams the admin logs with messages.

I'm not sure what the best way to fix this is, or if it even needs to be fixed given this only happens for low oxygen percentages. Maybe:
- remove the `respirator.Suffocating` bool
- replace logging condition with `SuffocationCycles == 2` and `>=2` in `TakeSuffocationDamage` and `StopSuffocation, respectively
- update `SuffocationCycles` **after** calling `TakeSuffocationDamage` or `StopSuffocation`.

That way it will probably only log it once per suffocation instance? Though the downside is then it won't always log when people are taking damage due to low-oxygen ratios.